plaso/parsers/gdrive_synclog.py

```diff
@@ -0,0 +1,197 @@
+    """Parses a logline and store appropriate attributes.
+
+    Args:
+      parser_mediator (ParserMediator): mediates interactions between parsers
+          and other components, such as storage and dfvfs.
+      structure (pyparsing.ParseResults): structure of tokens derived from
+          a line of a text file.
+    """
+    # Some hacks follow: our log source uses a different datetime format than
+    # dfdatetime does, and there appear to be some pending changes surrounding
+    # is_local_time and converting to UTC (see
+    # https://github.com/log2timeline/dfdatetime/issues/47, possibly others).
+    # As such, I'm hacking around things with some string reformats to get it
+    # working for a first-pass, and will take direction from Joachim when I have
+    # more context on the ongoing work/best practices.
+    year, month, day_of_month, hours, minutes, seconds, milliseconds, tz = (
+        structure.date_time)
+
+    # TimeElementsInMilliseconds appears not to support local timezones, yet,
+    # and time_elements.CopyFromDateTimeString requires a different format.
+    # Initialize it without the TZ, then convert to a TZ-aware value with
+    # a formatting hack and CopyFromDateTimeString.
+    dfdatetime_obj = dfdatetime_time_elements.TimeElementsInMilliseconds(
+        (year, month, day_of_month, hours, minutes, seconds, milliseconds))
+
+    dt_hack = ('{0:d}-{1:02d}-{2:02d} {3:02d}:'
+               '{4:02d}:{5:02d}.{6:03d}{7!s}:{8!s}').format(
+                   year, month, day_of_month, hours, minutes, seconds,
+                   milliseconds, tz[0:3], tz[3:5])
+    try:
+      dfdatetime_obj.CopyFromDateTimeString(dt_hack)
+    except ValueError:
+      parser_mediator.ProduceExtractionError(
+          'invalid date time value: {0!s}'.format(structure.date_time))
+      return
+
+    event_data = GoogleDriveSyncLogEventData()
+    event_data.log_level = structure.log_level
+    event_data.pid = structure.pid
+    event_data.thread = structure.thread
+    event_data.source_code = structure.source_code
+    # Replace newlines with spaces in structure.message to preserve output.
+    event_data.message = structure.message.replace('\n', ' ')
+
+    event = time_events.DateTimeValuesEvent(
+        dfdatetime_obj, definitions.TIME_DESCRIPTION_ADDED)
+
+    parser_mediator.ProduceEventWithEventData(event, event_data)
+
+  def ParseRecord(self, parser_mediator, key, structure):
+    """Parses a log record structure and produces events.
+
+    Args:
+      parser_mediator (ParserMediator): mediates interactions between parsers
+          and other components, such as storage and dfvfs.
+      key (str): identifier of the structure of tokens.
+      structure (pyparsing.ParseResults): structure of tokens derived from
+          a line of a text file.
+
+    Raises:
+      ParseError: when the structure type is unknown.
+    """
+    # Note: this may be overcomplicated/unnecessary; I'm mostly cloning SkyDrive
```

Comment:
Please remove the comment. If you feel this needs to be changed I opt to add an issue and address the change for all affected parsers.

Reply:
Done -- this may have been a misunderstanding w.r.t. how key types worked on my part, IIUC things are fine here as-is.